src/core/vectortile/qgsvectortilewriter.cpp

```diff
@@ -35,6 +35,48 @@
 
 QgsVectorTileWriter::QgsVectorTileWriter()
 {
+  setCrs( QgsCoordinateReferenceSystem( mDefaultCrs ) );
+}
+
+
+bool QgsVectorTileWriter::setCrs( const QgsCoordinateReferenceSystem &crs )
+{
+  if ( crs.authid() == mDefaultCrs )
+  {
+    mScaleDenom0 = 559082264.0287178;
+    mTopLeft = QgsPointXY( -20037508.3427892, 20037508.3427892 );
+    mCrs = crs;
+    return true;
+  }
+
+  const QgsCoordinateTransform ct( QgsCoordinateReferenceSystem( "EPSG:4326" ), crs, mTransformContext );
+  QgsRectangle ce;
+  try
+  {
+    ce = ct.transformBoundingBox( crs.bounds() );
```

Comment:
I don't think that a tiling scheme for a CRS should be defined entirely by the bounds of the CRS defined in the EPSG database. The actual bounds of the root tile should be explicitly given by the user rather than auto-calculated from the bounds.

For example, for EPSG:4326, there is often a tile scheme which has two root tiles so that they cover the whole extent -180,180 and -90,90. Without letting user to define arbitrary `QgsTileMatrix`, this won't be possible...

Also, for some CRS (e.g. for polar regions) this transform of bounds may not give sensible results...